### PR TITLE
English Auction

### PR DESCRIPTION
**Pull request by fortysevenlabs**

### CODE REVIEW

File: contracts/NFT.solpp [hunk 1]
```diff
@@ -1,5 +1,4 @@
 // SPDX-License-Identifier: MIT
-
 pragma soliditypp ^0.8.0;
 
 import "./openzeppelin/token/ERC721/extensions/ERC721Enumerable.sol";
```
Comment (franz101) [inline contracts/NFT.solpp:4]: Can recommend the contract wizard from Oz:
https://wizard.openzeppelin.com/#erc721

**Review by franz101 — APPROVED** [review]
Looks good to me.
I have to run the tests and I could add a GitHub action to ru the tests on each commit.

File: contracts/NFT.solpp (hunk 1 above)
Comment (fortysevenlabs) [reply contracts/NFT.solpp:4]: 👍 The ERC721/IERC721 use in our contracts can be simplified significantly. For now, I just wanted to use something to create NFTs for the tests.

**Comment by fortysevenlabs** [comment]
I'm going to continue pushing changes to this PR and we can merge once we feel good about the English Auction contract. Similar we can add another PR for Dutch Auction.

**Comment by fortysevenlabs** [comment]
Added tests for `englishAuction.start()`. The overhead/complexity of this project is mostly on the vite/vuilder setup and workings.

File: contracts/NFT.solpp (hunk 1 above)
Comment (fortysevenlabs) [reply contracts/NFT.solpp:4]: Also, I don't like the approach of uploading the contracts we import to the project and would prefer to leverage npm install them but took this approach to resolve some of the setup issues.

**Comment by fortysevenlabs** [comment]
There are a bunch of repetitive calls in tests that we should move to a function or before/beforeEach if it makes sense.

**Pull request merged**